dcos/subcommand.py

```diff
@@ -324,6 +405,95 @@ def _find_virtualenv(bin_directory):
     return virtualenv_path
 
 
+def _move(src, dst):
+    """Move file from src to the file or directory dst and remove dst if failure
+
+    :param src: source file
+    :type src: str
+    :param dst: destination file or directory
+    :type dst: str
+    :returns: None
+    """
+
+    try:
+        shutil.move(src, dst)
+    except Exception as e:
+        logger.exception(e)
+        shutil.rmtree(dst)
+        raise
+
+
+def _install_with_binary(
+        package_name,
+        env_directory,
+        binary_cli):
+    """
+    :param package_name: the name of the package
+    :type package_name: str
+    :param env_directory: the path to the directory in which to install the
+                          package's binary_cli
+    :type env_directory: str
+    :param binary_cli: binary cli to install
+    :type binary_cli: str
+    :rtype: None
+    """
+
+    binary_url, kind = binary_cli.get("url"), binary_cli.get("kind")
+    content_hashes = binary_cli.get("contentHash")
+
+    try:
+        env_bin_dir = os.path.join(env_directory, BIN_DIRECTORY)
+
+        if kind == "executable":
+            binary_name = "dcos-{}".format(package_name)
+            with util.temptext() as file_tmp:
+                _, binary_tmp = file_tmp
+                urllib.request.urlretrieve(binary_url, binary_tmp)
```

Comment:
Did you look into using `requests` or even better our `dcos.http`? Is there are reason why you are using `urllib.request`?